packages/material-ui/src/Collapse/Collapse.js

```diff
@@ -94,15 +122,19 @@ const Collapse = React.forwardRef(function Collapse(props, ref) {
   };
 
   const handleEnter = normalizedTransitionCallback((node, isAppearing) => {
-    node.style.height = collapsedHeight;
+    // Set absolute position to get the size of collapsed content
+    wrapperRef.current.style.position = 'absolute';
+    node.style[size] = collapsedSize;
 
     if (onEnter) {
       onEnter(node, isAppearing);
     }
   });
 
   const handleEntering = normalizedTransitionCallback((node, isAppearing) => {
-    const wrapperHeight = wrapperRef.current ? wrapperRef.current.clientHeight : 0;
+    const wrapperSize = wrapperRef.current[isHorizontal ? 'clientWidth' : 'clientHeight'];
```

Comment:
This used to be defensive. I would consider this an issue with the passed `component` but we're probably safer by keeping it defensive.

Though we might consider warning in the future. For this PR let's keep the defensive checks to make the tests pass.